contrib/annotations/src/main/java/io/opentelemetry/contrib/annotations/WithSpan.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.annotations;
```

Comment:
consider to use "io.opentelemetry.contrib.auto.annotations" to specify that they work with auto instrumentation.

Reply:
An interesting question is whether we would have another package for common, non-auto-related annotations potentially (in the future, that is).

Reply:
Based on my understanding this annotation will not be able to work without some kind of bytecode manipulation. So it is important to call that for the users and let them know that this should not be used if they are not sure if the final application will use byte-code manipulation.

In the same notes probably third-party integrations should not use this.

Reply:
I have added this notice in javadoc. Please review.

Reply:
@bogdandrutu Fair enough, lets take this route.